pkg/cmd/crdconversionwebhook/converter/converter.go

```diff
@@ -0,0 +1,89 @@
+package converter
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/console-operator/pkg/api"
+)
+
+func convertCRD(Object *unstructured.Unstructured, toVersion string) (*unstructured.Unstructured, metav1.Status) {
+	convertedObject := Object.DeepCopy()
+	fromVersion := Object.GetAPIVersion()
+
+	if toVersion == fromVersion {
+		return nil, statusErrorWithMessage("conversion from a version to itself should not call the webhook: %s", toVersion)
+	}
+
+	switch Object.GetAPIVersion() {
+	case "console.openshift.io/v1alpha1":
+		switch toVersion {
+		case "console.openshift.io/v1":
+			klog.Infof("converting %q object from 'console.openshift.io/v1alpha1' into 'console.openshift.io/v1'", convertedObject.GetName())
+			convertPlugins(convertedObject)
+		default:
+			return nil, statusErrorWithMessage("unexpected conversion version %q", toVersion)
+		}
+	default:
+		return nil, statusErrorWithMessage("unexpected conversion version %q", fromVersion)
+	}
+	return convertedObject, statusSucceed()
+}
+
+func convertPlugins(convertedObject *unstructured.Unstructured) (*unstructured.Unstructured, error) {
```

Comment:
I haven't deeply reviewed conversion logic.  Might it be easier to do `runtime.FromUnstructured` to get the original as a v1alpha1.ConsolePlugin and create a fresh v1.ConsolePlugin and directly set the various fields?  I won't stop this if the team prefers it, I just wonder if it will improve readability.